core/src/main/java/com/graphhopper/routing/util/BikeCommonFlagEncoder.java

```diff
@@ -550,7 +564,8 @@ void collect( OSMWay way, double wayTypeSpeed, TreeMap<Double, Integer> weightTo
     {
         String service = way.getTag("service");
         String highway = way.getTag("highway");
-        if (way.hasTag("bicycle", "designated") || way.hasTag("bicycle", "official"))
+        if ( ((highway!=null) && !highway.equals("steps")) && 
+              (way.hasTag("bicycle", "designated") || way.hasTag("bicycle", "official")) )
```

Comment:
if it is designated for bike then we do not need this steps check IMO? Also why is this related to segregated?